runtime/bindings/python/tests/vm_test.py

```diff
@@ -106,6 +107,27 @@ def test_add_scalar_new_abi(self):
     logging.info("result: %s", result)
     self.assertEqual(result, 11)
 
+  def test_mmap(self):
+    binary = iree.compiler.compile_str(
+        """
+        func.func @add_scalar(%arg0: i32, %arg1: i32) -> i32 {
+          %0 = arith.addi %arg0, %arg1 : i32
+          return %0 : i32
+        }
+        """,
+        target_backends=iree.compiler.core.DEFAULT_TESTING_BACKENDS)
+    with tempfile.NamedTemporaryFile() as tf:
+      tf.write(binary)
+      tf.flush()
+      m = iree.runtime.VmModule.mmap(self.instance, tf.name)
```

Comment:
> May need some tweaking for Windows

Indeed. https://github.com/openxla/iree/actions/runs/5275962562/jobs/9542060597#step:9:2321

```
ERROR: test_mmap (__main__.VmTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\a\iree\iree\runtime\bindings\python\tests\vm_test.py", line 122, in test_mmap
    m = iree.runtime.VmModule.mmap(self.instance, tf.name)
PermissionError: [Errno 13] Permission denied: 'C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\tmpr4u5ldjo'
```

Maybe related to https://github.com/openxla/iree/issues/13148#issuecomment-1520924831
> See https://stackoverflow.com/a/23212515 - NamedTemporaryFile creates and opens the file, and the file cannot be opened again... on Windows (it can be opened again on Unix).

Reply:
I've added a [fix](https://github.com/nod-ai/SHARK/blob/189afec2a6130b7bc0fb0188539c576b4bc924f3/shark/iree_utils/compile_utils.py#L357-L364) for that in SHARK to deal with this very issue - perhaps that can be used.

But the other issue in Windows which I bumped into is mentioned [here](https://github.com/nod-ai/SHARK/pull/1540#issuecomment-1593397675) - it's API related instead of the use case.

CC: @powderluv